lib/core/request.js

```diff
@@ -185,6 +201,10 @@ class Request {
     clearTimeout(this[kTimeout])
     this[kTimeout] = null
 
+    if (typeof this[kHandler].onComplete !== 'function') {
+      throw new InvalidArgumentError('invalid onComplete method')
+    }
+
     this[kHandler].onComplete(trailers)
   }
 
```

Comment:
I don't think you need to check here. Just check in the request constructor.

Reply:
Alright. I'll do the necessary changes and update the PR

Reply:
So I have reverted the changes to the existing tests. But it seems like moving the check to the request constructor is breaking a few tests as some of the handlers don't have all the methods (like `onComplete` ).

Reply:
@ronag I have updated the PR but I'm not sure how to proceed with the check in the request constructor. Can you please help here?

Reply:
Put the checks in the constructor and fix the tests that fail due to missing handlers?

Reply:
Okay. I was under the impression that it wasn't necessary for all the handler methods to be implemented. :sweat_smile:

Reply:
@ronag , it seems that @mcollina has already approved this. Are the changes to the constructor still required?

If yes, the failing tests would need a change in them. But as you suggested earlier, we should avoid changing existing tests. 

Also, I would like a bit more clarity. If I add it to the constructor, it would become mandatory to implement all the handler methods. Is this what is intended?

Reply:
> Are the changes to the constructor still required?

Yes

> If yes, the failing tests would need a change in them. But as you suggested earlier, we should avoid changing existing tests.

Try to avoid but if it's necessary we should change them.

> Is this what is intended?

Yes

Reply:
I have put the checks in the constructor and fixed the tests.